### PR TITLE
TS-4629: Fix use of invalid iterator in ServerSessionPool::purge.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
Add postfix increment to IPHashTable::iterator.

### CODE REVIEW

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/296/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/400/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/401/ for details.

File: proxy/http/HttpSessionManager.cc [hunk 1]
```diff
@@ -55,9 +55,10 @@ ServerSessionPool::ServerSessionPool() : Continuation(new_ProxyMutex()), m_ip_po
...
 ServerSessionPool::purge()
 {
-  for (IPHashTable::iterator last = m_ip_pool.end(), spot = m_ip_pool.begin(); spot != last; ++spot) {
-    spot->do_io_close();
-  }
+  // @c do_io_close can free the instance which clears the intrusive links and breaks the iterator.
+  // Therefore @c do_io_close is called on a post-incremented iterator.
+  for (IPHashTable::iterator last = m_ip_pool.end(), spot = m_ip_pool.begin(); spot != last ; spot++->do_io_close())
+    ; // empty
   m_ip_pool.clear();
```
Comment (dragon512) [inline proxy/http/HttpSessionManager.cc]: just a thought. As at this point the collection should be empty? do want to add an assert that it is vs calling clear() as calling clear() on an empty collection be a no-opt or a sign of a bug?

**Pull request closed without merging**